myriad-scheduler/src/main/java/org/apache/myriad/Main.java

```diff
@@ -76,16 +76,16 @@
   private static Injector injector;
 
   public static void initialize(Configuration hadoopConf, AbstractYarnScheduler yarnScheduler, RMContext rmContext,
-                                InterceptorRegistry registry) throws Exception {
-    ObjectMapper mapper = new ObjectMapper(new YAMLFactory());
-    MyriadConfiguration cfg = mapper.readValue(Thread.currentThread().getContextClassLoader().getResource(
-        "myriad-config-default.yml"), MyriadConfiguration.class);
+                                InterceptorRegistry registry) throws Exception {    
+    initConfiguration(hadoopConf, yarnScheduler, rmContext, registry);
 
-    MyriadModule myriadModule = new MyriadModule(cfg, hadoopConf, yarnScheduler, rmContext, registry);
+    new Main().run(injector.getInstance(MyriadConfiguration.class));
+  }
+  
+  private static void initConfiguration(Configuration conf, AbstractYarnScheduler scheduler, RMContext context, InterceptorRegistry registry) {
+    MyriadModule myriadModule = new MyriadModule("myriad-config-default.yml", conf, scheduler, context, registry);
     MesosModule mesosModule = new MesosModule();
     injector = Guice.createInjector(myriadModule, mesosModule, new WebAppGuiceModule());
-
-    new Main().run(cfg);
   }
```

Comment:
No, just thought it would be cool to modularize a bit.  I can change it back if you'd prefer.